src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.servicehandler.FruitOperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final Map<FruitTransaction.Operation, FruitOperationHandler> operationHandlers;
+
+    public ShopServiceImpl(Map<FruitTransaction.Operation,
+            FruitOperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            FruitOperationHandler handler = operationHandlers.get(transaction.getOperation());
+            if (handler != null) {
+                try {
+                    handler.handle(transaction);
+                } catch (Exception e) {
+                    System.err.println("Error processing transaction " + transaction
+                            + ": " + e.getMessage());
+                }
```

Comment:
Consider catching more specific exceptions instead of the generic `Exception`. This will help in identifying and handling different types of errors more effectively. If specific exceptions are not known, consider logging the stack trace for better debugging.